spec/unit/util/log_spec.rb

```diff
@@ -256,9 +289,108 @@ def log_notice(message)
       log = logs[0]
       expect(log.message).to_not match('/log_spec.rb')
       expect(log.backtrace).to be_a(Array)
+      expect(log.backtrace.first).to match('/log_spec.rb')
+    end
+
+    it "backtrace has interleaved PuppetStack when logging ParseErrorWithIssue" do
+      logs = []
+      destination = Puppet::Test::LogCollector.new(logs)
+      Puppet::Util::Log.newdestination(destination)
+
+      Puppet[:trace] = true
+      Puppet::Util::Log.with_destination(destination) do
+        PuppetStackCreator.new.run(Puppet::ParseErrorWithIssue)
+      end
+      Puppet[:trace] = false
+
+      # Normal logging exepectations are still true
+      expect(logs.size).to eq(1)
+      log = logs[0]
+      expect(log.message).to_not match('/log_spec.rb')
+      expect(log.backtrace[0]).to match('/log_spec.rb')
```

Comment:
nitpick, but above you use `.first` instead of `[0]`, if you wanna be consistent